applications/services/gui/canvas.c

```diff
@@ -298,21 +298,37 @@ void canvas_draw_icon_animation(
     Canvas* canvas,
     int32_t x,
     int32_t y,
+    int32_t width_scale,
+    int32_t height_scale,
```

Comment:
unfortunately we cannot change this, as this function is exposed to public api and must remain compatible across firmwares and also not break backwards compatibility. could make a separate function for this. to avoid code duplication, this current function could be made static and add a `_` in front, then make 2 new functions, one with original signature and one with scale arguments, that just call the common static function. the new function could be called ..._ex() to indicate the extended functionality

Reply:
Good info thank you I will into applying that now 👍

Reply:
Good to go. Thank you!